lesson11/main.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+const endLoading = () => wrapper.remove();
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch(error) {
+      throw new Error(`データの取得に失敗しました：${error}`);
+   }
+};
+
+async function init() {
+   startLoading();
+   try {
+      const result = await fetchData();
+      createListElements(result);
+   } catch(e) {
+      const div = document.createElement("div");
+      lists.appendChild(div);
+      div.textContent = 'データの取得に失敗しました';
+      console.log('データが取得に失敗しました');
```

Comment:
I think it would be better to use the "e" argument of catch (as in fetchData).
If an error occurs in fetchData, the catch(e) in init() will contain the error thrown, and we want to display it.
For any other errors that occur, I want to display the details of the error and the fact that an error occurred.

When an error occurs in fetchData, the catch(e) in init() will contain the error that was thrown, and I want to display it.


I've written down my thoughts.
[codesandbox](https://codesandbox.io/s/unruffled-euclid-buosp?file=/main.js)